liberty-managed/src/main/java/io/openliberty/arquillian/managed/WLPManagedContainer.java

```diff
@@ -1249,42 +1244,32 @@ private void throwDeploymentException(String applicationName) throws DeploymentE
-            StringBuilder sb = new StringBuilder();
-            sb.append("Failed to deploy ")
-                  .append(applicationName)
-                  .append(" on ")
-                  .append(containerConfiguration.getServerName());
-
-            Throwable ffdcXChain = getFfdcWithNestedCauseChain(applicationName);
-
-            if (bestLine.contains("DefinitionException")) {
-               log.finest("DefinitionException found in line" + bestLine + " of file " + messagesFilePath);
-               DefinitionException cause = new javax.enterprise.inject.spi.DefinitionException(bestLine, ffdcXChain);
-               throw new DeploymentException(sb.toString(), cause);
-            } else if (bestLine.contains("DeploymentException") ||
-                       bestLine.contains("InconsistentSpecializationException") ||
-                       bestLine.contains("UnserializableDependencyException")) {
-               /*
-                * The CDI specification allows an implementation to throw a subclass of
-                * javax.enterprise.inject.spi.DeploymentException. Weld has three types
-                * such exceptions:
-                *  - org.jboss.weld.exceptions.DeploymentException
-                *  - org.jboss.weld.exceptions.InconsistentSpecializationException
-                *  - org.jboss.weld.exceptions.UnserializableDependencyException
-                */
-               log.finest("DeploymentException found in line" + bestLine + " of file " + messagesFilePath);
-               javax.enterprise.inject.spi.DeploymentException cause = new javax.enterprise.inject.spi.DeploymentException(bestLine, ffdcXChain);
-               throw new DeploymentException(sb.toString(), cause);
-            } else {
-               /*
-                * Application failed to deploy due to some other exception.
-                */
-               String exceptionFound = bestLine.substring(bestLine.indexOf("The exception message was: ") + 27);
-               sb.append(": ");
-               sb.append(exceptionFound);
-               log.finest("A exception was found in line " + bestLine + " of file " + messagesFilePath);
-               throw new DeploymentException(sb.toString(), ffdcXChain);
-            }
+           log.finest("A exception was found in line " + bestLine + " of file " + messagesFilePath);
+
+           // Attempt to find the exception that occurred on the server
+           FFDCExceptionLocator ffdcLocator = new FFDCExceptionLocator(getLogsDirectory());
+           CDILogExceptionLocator cdiLocator = new CDILogExceptionLocator();
+           
+           long deploymentTime = archiveDeployTimes.get(applicationName);
+           
+           Throwable serverException = ffdcLocator.getException(applicationName, bestLine, deploymentTime);
+           
+           if (serverException == null) {
+               serverException = cdiLocator.getException(applicationName, bestLine, deploymentTime);
+           }
```

Comment:
Do we need a null check here if `serverException` if we didn't find any exceptions?

Reply:
If we don't find an exception here, then we'll pass `null` as the cause to the eventual `DeploymentException` (line 1271) which should be fine.